qiskit/transpiler/passes/basis/basis_translator.py

```diff
@@ -225,68 +205,77 @@ def run(self, dag):
         # Replace source instructions with target translations.
 
         replace_start_time = time.time()
-        for node in dag.op_nodes():
-            node_qargs = tuple(qarg_indices[bit] for bit in node.qargs)
-            qubit_set = frozenset(node_qargs)
 
-            if node.name in target_basis:
-                continue
-            if (
-                node_qargs in self._qargs_with_non_global_operation
-                and node.name in self._qargs_with_non_global_operation[node_qargs]
-            ):
-                continue
+        def apply_translation(dag):
+            for node in dag.op_nodes():
+                node_qargs = tuple(qarg_indices[bit] for bit in node.qargs)
+                qubit_set = frozenset(node_qargs)
+                if node.name in target_basis:
+                    if isinstance(node.op, ControlFlowOp):
+                        flow_blocks = []
+                        for block in node.op.blocks:
+                            dag_block = circuit_to_dag(block)
+                            flow_dag_block = apply_translation(dag_block)
```

Comment:
Here's a case where the mutation makes things a bit unclear, and why we might be better using the return value for something else.  After this call, `flow_dag_block is dag_block` (i.e. they're the exact same Python object), but the call makes it look like `apply_translation` creates a new object.  If we instead use the return value to tell us whether we actually modified `dag_block`, we can potentially avoid the subsequent `dag_to_circuit` and `replace_blocks` calls later, for no additional computational cost.

Reply:
ok